### PR TITLE
Users now able to set the send by email setting by themselves

### PR DESCRIPTION
**Pull request by CrystalPea**
Story ticket: https://www.pivotaltracker.com/story/show/171254127

Related API PR: https://github.com/alphagov/notifications-api/pull/2729 (they should go out more-less together)

I also did some test work:
- added a function to look for element by tag and partial text, to avoid finding tag by order in the table, as that was quite brittle.
- updated Beautiful Soup syntax in our code where I found we were using old, deprecated forms. Thanks Katie for spotting! 💖 

### Before we set contact details:
Screenshot 2020-02-25 at 17 16 04

Screenshot 2020-02-25 at 13 36 38

### After we set contact details:
Screenshot 2020-02-25 at 13 40 11

Screenshot 2020-02-25 at 13 37 25

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -620,3 +620,7 @@ def assert_url_expected(actual, expected):
                 'Expected redirect: {}\n'
                 'Actual redirect: {}'
             ).format(expected, actual)
+
+
+def find_element_by_tag_and_partial_text(page, tag, string):
+    return [e for e in page.findAll(tag) if string in e.text][0]
```
Comment (klssmith) [inline tests/__init__.py]: The camel cased Beautiful Soup names are deprecated and only included for backwards compatibility, so this should become `find_all` and `findNextSibling` should now be `find_next_sibling` https://www.crummy.com/software/BeautifulSoup/bs4/doc/#method-names

Comment (CrystalPea) [reply tests/__init__.py]: Great shout, thanks, google must have linked me to old documentation 😆

File: app/templates/views/service-settings.html [hunk 2]
```diff
@@ -110,6 +97,16 @@ <h1 class="heading-medium">Settings</h1>
           )}}
         {% endcall %}
 
+        {% call settings_row(if_has_permission='email') %}
+          {{ text_field('Send files by email') }}
+          {{ text_field(current_service.contact_link if current_service.contact_link else "Not set up yet", truncate=true) }}
```
Comment (karlchillmaid) [inline app/templates/views/service-settings.html]:
```suggestion
          {{ text_field(current_service.contact_link if current_service.contact_link else "Not set up", truncate=true) }}
```

File: tests/app/main/views/test_service_settings.py [hunk 3]
```diff
@@ -180,11 +186,45 @@ def test_organisation_name_links_to_org_dashboard(
...
 
-    org_row = response.select('tr')[18]
+    org_row = find_element_by_tag_and_partial_text(response, tag='tr', string='Organisation')
     assert org_row.find('a')['href'] == url_for('main.organisation_dashboard', org_id=ORGANISATION_ID)
     assert normalize_spaces(org_row.find('a').text) == 'Test Organisation'
 
 
+@pytest.mark.parametrize('service_contact_link,expected_text', [
+    ('contact.me@gov.uk', 'Send files by email contact.me@gov.uk Manage'),
+    (None, 'Send files by email Not set up yet Manage'),
```
Comment (karlchillmaid) [inline tests/app/main/views/test_service_settings.py]:
```suggestion
    (None, 'Send files by email Not set up Manage'),
```

File: app/templates/views/service-settings/send-files-by-email.html [hunk 4]
```diff
@@ -7,20 +7,22 @@
...
       {{ page_header(
-        '{} contact details for ‘Download your document’ page'.format('Change' if 'upload_document' in current_service.permissions else 'Add'),
+        'Send files by email',
         back_link=url_for('main.service_settings', service_id=current_service.id)
       ) }}
       <p>
-        When you send users a document to download, you need to include the contact details for your service
-        on the download page. This is so users can contact you if there’s a problem (for example,
-        if the link to download the document has expired).
+        Our <a href={{ url_for('main.documentation') }}>API documentation</a> explains how to upload a file, then send your users an email with a link to download it.
```
Comment (karlchillmaid) [inline app/templates/views/service-settings/send-files-by-email.html]:
```suggestion
        This is an API-only feature.
        </p>
        <p>
        To send a file by email, follow the instructions in our <a href={{ url_for('main.documentation') }}>API documentation</a>.
```

File: app/templates/views/service-settings/send-files-by-email.html [hunk 5]
```diff
@@ -7,20 +7,22 @@
...
+      </p>
+      <h2 class="heading-medium">{% if contact_details %}Change{% else %}Add{% endif %} contact details to the file download page</h2>
+      <p>
+        You need to include contact details for your service so your users can contact you if there’s a problem. For example, if the link to download the file you sent them has expired.
```
Comment (karlchillmaid) [inline app/templates/views/service-settings/send-files-by-email.html]:
```suggestion
        You need to include contact details for your service so your users can get in touch if there’s a problem. For example, if the link to download the file you sent them has expired.
```

File: app/templates/views/service-settings/send-files-by-email.html [hunk 6]
```diff
@@ -7,20 +7,25 @@
...
       <p>
-        When you send users a document to download, you need to include the contact details for your service
-        on the download page. This is so users can contact you if there’s a problem (for example,
-        if the link to download the document has expired).
+        This is an API-only feature.
+        </p>
+        <p>
+        To send a file by email, follow the instructions in our <a href={{ url_for('main.documentation') }}>API documentation</a>.
+      </p>
+      <h2 class="heading-medium">{% if contact_details %}Change{% else %}Add{% endif %} contact details to the file download page</h2>
```
Comment (karlchillmaid) [inline app/templates/views/service-settings/send-files-by-email.html]:
```suggestion
      <h2 class="heading-medium">{% if contact_details %}Change contact details for{% else %}Add contact details to{% endif %} the file download page</h2>
```

**Pull request merged**